src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1122,6 +1122,25 @@ public function getPowType(Expr $left, Expr $right, callable $getTypeCallback):
 		$leftType = $getTypeCallback($left);
 		$rightType = $getTypeCallback($right);
 
+		if ($leftType instanceof MixedType || $rightType instanceof MixedType) {
+			return new BenevolentUnionType([
+				new FloatType(),
+				new IntegerType(),
+			]);
+		}
+
+		$object = new ObjectWithoutClassType();
+		if (
+			!$leftType instanceof NeverType &&
+			!$rightType instanceof NeverType &&
```

Comment:
I added these 2 never guard conditions, to not regress existing integer-range-math tests in https://github.com/phpstan/phpstan-src/blob/ff0d36e26667cac4a18f26576b0cd2fe96d7faf6/tests/PHPStan/Analyser/data/math.php#L157-L158